util/src/main/resources/META-INF/faces-config.xml

```diff
@@ -16,17 +16,18 @@
 		</system-event-listener>
 	</application>
 	<factory>
+		<application-factory>com.liferay.faces.util.application.internal.ApplicationFactoryUtilImpl</application-factory>
 		<partial-view-context-factory>com.liferay.faces.util.context.internal.PartialViewContextFactoryImpl</partial-view-context-factory>
 		<faces-context-factory>com.liferay.faces.util.context.internal.FacesContextFactoryUtilImpl</faces-context-factory>
 		<render-kit-factory>com.liferay.faces.util.render.internal.RenderKitFactoryUtilImpl</render-kit-factory>
 		<factory-extension>
 			<util:browser-sniffer-factory>com.liferay.faces.util.client.internal.BrowserSnifferFactoryImpl</util:browser-sniffer-factory>
-			<util:component-resource-factory>com.liferay.faces.util.application.internal.ComponentResourceFactoryImpl</util:component-resource-factory>
 			<util:faces-context-helper-factory>com.liferay.faces.util.context.internal.FacesContextHelperFactoryImpl</util:faces-context-helper-factory>
 			<util:faces-request-context-factory>com.liferay.faces.util.context.internal.FacesRequestContextFactoryImpl</util:faces-request-context-factory>
 			<util:faces-url-encoder-factory>com.liferay.faces.util.render.internal.FacesURLEncoderFactoryImpl</util:faces-url-encoder-factory>
 			<util:jsp-adapter-factory>com.liferay.faces.util.jsp.internal.JspAdapterFactoryImpl</util:jsp-adapter-factory>
 			<util:message-context-factory>com.liferay.faces.util.context.internal.MessageContextFactoryImpl</util:message-context-factory>
+			<util:resource-dependency-handler-factory>com.liferay.faces.util.application.internal.ResourceDependencyHandlerFactoryImpl</util:resource-dependency-handler-factory>
```

Comment:
In a perfect world `ResourceDependencyHandler` would be like `ResourceHandler` and be implemented as a delegation chain. Currently the Javadoc requires that it be implemented that way, but we could remove need for a `ResourceDependencyHandlerFactory` by detecting the `ResourceDependencyHandler` class in the `faces-config.xml`:

```
<!-- ... -->
    <application-extension>
        <util:resource-dependency-handler>
            com.liferay.faces.util.application.internal.ResourceDependencyHandlerImpl
        </util:resource-dependency-handler>
    </application-extension>
</application>
```

Once the class is detected, the `ApplicationUtilImpl` or `UIViewRootUtilImpl` could instantiate it with reflection similar to `FactoryFinder` and put it in a delegation chain (probably just a `List<ResourceDependencyHandler>` would be fine).